### PR TITLE
Rename folders for clarity [AEK-269]

### PR DESCRIPTION
**Pull request by aliphys**

### CODE REVIEW

**Review by per1234 — CHANGES_REQUESTED** [review]
Please resolve the sketch name mismatch at `examples/Firmware/Flasher.ino`.

**Review by per1234 — CHANGES_REQUESTED** [review]
The examples paths in the "Compile Examples" CI workflow must be updated at the same time these examples are moved:
- https://github.com/arduino-libraries/ArduinoMotorCarrier/blob/6e8550286713c5a30d053aeca41294644cdea5a7/.github/workflows/compile-examples.yml#L91
- https://github.com/arduino-libraries/ArduinoMotorCarrier/blob/6e8550286713c5a30d053aeca41294644cdea5a7/.github/workflows/compile-examples.yml#L97
- https://github.com/arduino-libraries/ArduinoMotorCarrier/blob/6e8550286713c5a30d053aeca41294644cdea5a7/.github/workflows/compile-examples.yml#L114

File: .github/workflows/compile-examples.yml [hunk 1]
```diff
@@ -112,6 +113,8 @@ jobs:
             ${{ matrix.libraries }}
           sketch-paths: |
             - examples/Flasher
+            - examples/MKRMotorCarrier/Test
```
Comment (per1234) [inline .github/workflows/compile-examples.yml:116]: I'm wondering about this. This causes the sketch to be compiled for all boards, but I would expect that the sketches under `examples/MKRMotorCarrier` would be only for use with MKR boards. I don't know much about these example sketches. Is this sketch intended to be universally applicable to all boards that are compatible with the library?

Comment (aliphys) [reply .github/workflows/compile-examples.yml]: I have removed it from the workflow for now.

**Review by per1234 — APPROVED** [review]

File: examples/MKRMotorCarrier/Test/Test.ino [hunk 2]
```diff
@@ -46,7 +46,7 @@ void setup() {
   // This way, you can program the motor to reach a certain position or velocity without any further intervention.
   // The PID can be carefully tuned if a particular profile is needed.
   pid1.setControlMode(CL_POSITION);
-  pid1.setGains(25, 0, 3);
+  pid1.setGains(25.0f, 0.0f, 3.0f);
```
Comment (aliphys) [inline examples/MKRMotorCarrier/Test/Test.ino:49]: pid1.setGains() works with `float `as the input datatype (not `int`). Thanks @Rocketct 😄

Comment (aliphys) [reply examples/MKRMotorCarrier/Test/Test.ino:49]: @per1234 Test.ino should compile correctly now 👍

**Review by Rocketct — APPROVED** [review]

**Review by Rocketct — COMMENTED** [review]
you added ` examples/NanoMotorCarrier` and `examples/MKRMotorCarrier` but if you don't change the folder name too on the file system, the github action will fail when the will try to find these path

**Comment by aliphys** [comment]
@Rocketct can you please help me change the required files required for the GitHub Action? I am learning GitHub actions as we go along. 😃

**Comment by Rocketct** [comment]
https://github.com/arduino-libraries/ArduinoMotorCarrier/tree/master/examples check the names here as you can see here the folder name are MKR and Nano, in your PR you changed the name in the github action as examples/NanoMotorCarrier and examples/MKRMotorCarrier you can do two action:

- modify the folder name and use the new naming
- get back the name to `MKR` and `Nano`

one think @aliphys , did you discused these changes with @ErnestoELC ?
i don't know if this could be a problem with matlab file system management

**Comment by aliphys** [comment]
@Rocketct I have made the changes in commit https://github.com/arduino-libraries/ArduinoMotorCarrier/pull/36/commits/5dd1e202924dfe93b1aa6537cdcca37609156fec .
Good question about the MATLAB integration, I will ask in our stand-up tomorrow. 👍

**Comment by Rocketct** [comment]
@aliphys `This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.`

@aliphys  please double check apparently that commit was not merged, or is not present on master, make that changes and add it to this PR.

but first of all talk with @ErnestoELC this repository cannot be modified so lightly as the other, second and important point because we are using an implicit cast can you and @ErnestoELC check if working or not, because could happened stranger things so we should be sure that is working roperly.

**Comment by aliphys** [comment]
@Rocketct just spoke with MW, changing the `examples` folder and its content does not influence the Hardware Support Packages.

**Comment by Rocketct** [comment]
okk perfect so make the changes on the folders name and after the test could be merged

**Comment by Rocketct** [comment]
ah okk looks like you have made it let me check if the github action pass

**Comment by Rocketct** [comment]
@per1234 ciao Per can you check the github action? it fails when start to compile the examples

**Comment by per1234** [comment]
@Rocketct everything looks as expected to me. There are some compilations that fail, but those are true results.

You will see some of these if you look in the logs:
```
Compiling sketch: examples/MKRMotorCarrier/Motor_test_encoder
Compiling previous version of sketch to determine memory usage change
Compiling sketch: examples/MKRMotorCarrier/Motor_test_encoder
  
  Error during build: opening sketch: no valid sketch found in /home/runner/work/ArduinoMotorCarrier/ArduinoMotorCarrier/examples/MKRMotorCarrier: missing /home/runner/work/ArduinoMotorCarrier/ArduinoMotorCarrier/examples/MKRMotorCarrier/MKRMotorCarrier.ino
  
Error: Compilation failed
Change in flash: N/A
Change in RAM for global variables: N/A
```
The failure of the "Compiling previous version of sketch to determine memory usage change" compilation is normal and expected in this case where the sketch did not exist at that path in the PR's base ref. This second compilation is only done for the sake of the memory usage change report, which is purely informational. The failure of that compilation does not change the CI result. It will start working fine on the next PR after the sketches have been established at their new location.

So focus only on the parts of the logs where the first compilation failed. Those are the ones that cause the ❌ .

---
Note also that there were widespread sporadic GitHub Actions outages all day. My email inbox is full of notifications of spurious workflow failures. These were transient so they were usually passing after I re-ran the jobs. The results you see from this PR are from days ago, before that happened, but if you tried running the CI today then you might have encountered some of these failures. GitHub Actions seems to be increasingly unreliable lately, but at least they get it back up to normal fairly quickly.

**Comment by Rocketct** [comment]
@per1234 ah ok i though that these ones:
```
Compiling sketch: examples/MKRMotorCarrier/Motor_test_encoder
Compiling previous version of sketch to determine memory usage change
Compiling sketch: examples/MKRMotorCarrier/Motor_test_encoder
  
  Error during build: opening sketch: no valid sketch found in /home/runner/work/ArduinoMotorCarrier/ArduinoMotorCarrier/examples/MKRMotorCarrier: missing /home/runner/work/ArduinoMotorCarrier/ArduinoMotorCarrier/examples/MKRMotorCarrier/MKRMotorCarrier.ino
  
Error: Compilation failed
Change in flash: N/A
Change in RAM for global variables: N/A
```
where fails not that are ok so fine i think that we should focus on portenta's one, but i think that there is a mistake because as far as i remember this library should not be compatible with portenta, can you confirm @aliphys @ErnestoELC ?

**Comment by aliphys** [comment]
@per1234 I appreciate your explanation of the GH actions. 👍 

@Rocketct Neither boards are compatible with the Portenta, with no plan to provide compatibility in the near future. The Motor Carrier library is designed to work with the Nano 33 IoT (for the [Nano Motor Carrier](https://docs.arduino.cc/hardware/nano-motor-carrier) examples) and all the MKR boards (for the [MKR Motor Carrier](https://docs.arduino.cc/hardware/mkr-motor-carrier)). You can see the list of compatible boards on their relevant docs pages.

**Note:** Nano BLE 33 Sense compatibility is experimental at the moment.

**Comment by per1234** [comment]
I'm the one who added the compilation for Portenta H7 to the CI workflow. The reason is because this statement:
https://store.arduino.cc/portenta-h7
>MKR Headers | Use any of the existing industrial MKR shields on it

implied to me that the MKR form factor accessories are intended to be compatible with the Portenta. If the hardware is intended to be compatible then I would expect the software to be compatible as well.

**Comment by Rocketct** [comment]
@per1234 ah ok mmm for this particular case should not or better was not implemented the software support to this shield and i don't think we want it, but if mandatory we could see how to implement it

**Comment by per1234** [comment]
Well, I had to make some educated guesses as to which boards a library was intended to support when it came to setting up the compilation workflows. The point of the workflow is to find example/board combos that don't compile, so I couldn't use a failure to compile as an indication that a board should not be compiled for.

In cases where the CI is configured to compile for a board when there is no intention of  the library supporting that board, I'm always happy to correct the CI configuration. However, in that case I would also suggest the evaluation of whether the supported boards can be more clearly communicated to the user, since if it was unclear to me then it might well be unclear to others.

**Comment by Rocketct** [comment]
the portenta and the shield are hw incompatible

**Comment by per1234** [comment]
PR submitted to remove the Portenta compilations from the CI workflow: https://github.com/arduino-libraries/ArduinoMotorCarrier/pull/39

**Comment by aliphys** [comment]
> the portenta and the shield are hw incompatible
@Rocketct can you kindly clarify why they are HW incompatible? The Portenta does have all the pins for the MKR boards.

**Comment by Rocketct** [comment]
A1 and A2 cannot be used as analog out and they are used  to controls motors through the shield, so the shield and the Portenta cannot be used together, so there is no reason to fix the software because at last the hardware will not work than have no sense include it on the .github action because we don't want to check something that is no compatible and will never be fixed

**Comment by aliphys** [comment]
@Rocketct I have taken a look at the schematic. A1 and A2 pass through the bidirection voltage translator and then connect to IN2 and IN4 respectively. How does the lack of PWM influence the DAC capacilities?

**Comment by Rocketct** [comment]
are two different things, however in this case the problem is that from A0 to A4 are connected to a switch (analog switch) that allows to choose if use this pin as PXY_C or PXY, really the incompatibility born in the exactly moment when you decide to use the SPI, that i notice is not populated so there could be a chance to make it usable, but in general if you use alone the pins only as a pure analog there shouldn't be any problem, but if you use PXY and meanwhile you want to use PXY_C functionality this is not possible,because from an hw point of view you are allowed to use only one pad at time.

**Comment by aliphys** [comment]
@Rocketct If we use the labelled pins only for analog input (i.e. use the ADC), then there shouldn't be a HW blocker for using the Portenta with the MKR Motor Carrier, is this correct?
image

**Comment by Rocketct** [comment]
no @aliphys  that pin is conneted to sig INX pin, but AFAIK that connector is used to connect motors too rigth? if yes the sig signal shall send some signal to control the motor than analog out.
But looks the SPI have DNP so this means that the SPI is disconnected than this means that this could not affect the compatibility, than the product could works with portenta

**Comment by aliphys** [comment]
The INX (IN1-IN4) connectors are designated as analog inputs (so the user can attach external sensors) and not used to control the motors.

**Comment by Rocketct** [comment]
ok so this means that the problem does't exist

**Comment by aliphys** [comment]
That is great @Rocketct 😃 .
How much work would it take to to make the ArduinoMotorCarrier library compatible with the Portenta H7?

**Comment by Rocketct** [comment]
wait before this we want that is comaptible? what say the project owner?

**Pull request merged**